### ISSUE TITLE
HasMany sourceKey not working or not available

### ISSUE THREAD

**Issue by truginis** [RobinBuschmann/sequelize-typescript#479]
Not sure if i am doing something wrong but i can't get sourceKey to work for the life of me in a one-to-many association. The Employee model keeps using id column for the join. Code below. 

```typescript
@Table
export class Employee extends Model<Employee> {
  @Column({
    type: DataType.STRING,
    primaryKey: true
  })
  id: string;

  @Column({
    type: DataType.STRING,
    field: 'Employee_Name',
  })
  fullName: string;

  @HasMany(() => Deal, {
    sourceKey: 'Employee_Name', //tried with 'fullName' as well
  })
  deals: Deal[];
}

@Table
export class Deal extends Model<Deal> {
  @Column({
    primaryKey: true,
    type: DataType.INTEGER,
    field: 'ADF_PROPOSAL_NUMBER'
  })
  id: number;

  @ForeignKey(() => Employee)
  @Column({
    type: DataType.STRING,
    field: 'Legal_Name'
  })
  get assignedLegal(): string {
    const value = this.getDataValue('assignedLegal') || null;
    return !value ? null : value.trim();
  }

  @BelongsTo(() => Employee, 'assignedLegal')
  legalRep: Employee;
}
```

**Comment by truginis** [comment]
Quick update. Figure out how to make this work, but not sure if ```sourceKey``` should still be fixed/changed. 

Solution: Just add ```targetKey``` to the belongsTo. 

```typescript
 @BelongsTo(() => Employee, {foreignKey: 'assignedLegal', targetKey: 'fullName'})
  legalRep: Employee;
```

**Comment by RobinBuschmann** [comment]
@truginis It should also be working if you omit the explicit references to the foreign keys totally.
So remove `'assignedLegal'` from `@BelongsTo(() => Employee, 'assignedLegal')` and `sourceKey` from `@HasMany(() => Deal, {sourceKey: 'Employee_Name', //tried with 'fullName' as well})` (What I also notices is, that you should use the property key in sourceKey instead of the field name)

Hope this helps! (If you're happy with your/my suggested solution, can we close this issue then?)

**Comment by truginis** [comment]
@RobinBuschmann thanks for a quick reply! I truncated most of the model to make it easier to read on here. I have multiple associations and multiple foreignKeys so that's why I am explicitly referencing it. 

Everything is working fine now by just using targetKey. I tried using property key for the sourceKey as well, but after googling for a while i found an issue on the sequelize js that causes problems if the  property key is different than the field. Still not sure what sourceKey even does since i can change it to anything i want and it won't change the queries in any way.
